sources/engine/Stride.Engine/Engine/Startup/GameBuilder.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+
+namespace Stride.Engine.Startup;
+
+public interface IGameBuilder
+{
+    public ConfigurationManager Configuration { get; }
+    public IServiceCollection Services { get; }
+    public ILoggingBuilder Logging { get; }
+    public IAppEnvironment Environment { get; }
```

Comment:
ASP Core has IHostEnvironment, I created IAppEnvironment as counterpart.
I know we already have something similar, but that is only created later
after `new Game()` is called.

I wanted to highlight that something like this is needed as part of this GameBuilder already, for 2 reasons:

A) If reading from a file like appsettings.json for Configuration,
it needs to correctly get that file. ASP uses
IHostEnvironment.ContentRootFileProvider for this - we would need to check
if Stride's file reading stuff can already work here, without other dependencies etc.

B) Setting up the other parts of this builder usually already needs the 
the info if the application is running in debug vs release mode.
"Set up profiler only if we are in debug mode" etc.

Reply:
> Perhaps I'm taking liberties by expanding it beyond a purely network-centric definition.

Actually, that's exactly the opposite of how it is. It took me some time to understand that the word "host" as they're using it in the context of "generic host" kinda has nothing to do with networking, and is more something along the lines of "the framework that 'hosts' your application on the computer.

So... "there are no solutions, only tradeoffs". If you end up going with the standard "host" naming - and taking into account that many people get into programming through game development - it would probably be a good idea to add documentation explaining this. Might save quite a few ~helpdesk tickets~ discord questions.